Can no longer build golint

In https://github.com/dexidp/dex/commit/076cd774693247587e604d2ce9fd8ca71f68e6b1 the ability to do `make lint` broke.

```
go: finding github.com/dexidp/dex/vendor/golang.org/x/lint/golint latest
go: finding github.com/dexidp/dex/vendor/golang.org/x/lint latest
go: downloading github.com/dexidp/dex/vendor/golang.org/x/lint v0.0.0-20190802071240-d36e6c26eea5
go: extracting github.com/dexidp/dex/vendor/golang.org/x/lint v0.0.0-20190802071240-d36e6c26eea5
go: github.com/dexidp/dex/vendor/golang.org/x/lint@v0.0.0-20190802071240-d36e6c26eea5: parsing go.mod: unexpected module path "golang.org/x/lint"
go: error loading module requirements
make: *** [Makefile:76: bin/golint] Error 1
```

Switching back to the previous commit has this working correctly:
```
$ git checkout 7c1b4b3
Previous HEAD position was 076cd77 run 'go get -u; make revendor'
HEAD is now at 7c1b4b3 Merge pull request #1502 from dexidp/sr/fix-log-formatting-in-VerifyPassword-grpc
$ make lint
$ ls -l bin/golint
-rwxr-xr-x  1 venezia  staff  6300320 Aug  2 14:06 bin/golint
```

Upon further research, it appears that https://github.com/golang/lint/commit/d0100b6bd8b389f0385611eb39152c4d7c3a7905 is the source of the issue.  Reverting to the previous commit (there are no releases of this library) seems to fix the issue.  There is a larger discussion to be had by viewing https://github.com/golang/go/issues/30831 and https://github.com/golang/go/wiki/Resolving-Problems-From-Modified-Module-Path

I have a branch with the updated go.mod/go.sum that should alleviate this issue.  While go mod is patched to understand the situation better (I hope anyway) I think this will be OK since we're not relying on super fresh features being added to that library or grpc (the grpc library reverts to an older version)

--- a/Makefile
+++ b/Makefile
@@ -2,6 +2,7 @@ PROJ=dex
 ORG_PATH=github.com/dexidp
 REPO_PATH=$(ORG_PATH)/$(PROJ)
 export PATH := $(PWD)/bin:$(PATH)
+THIS_DIRECTORY:=$(shell dirname $(realpath $(lastword $(MAKEFILE_LIST))))
 
 VERSION ?= $(shell ./scripts/git-version)
 
@@ -73,7 +74,7 @@ bin/protoc-gen-go:
 	@go install -v $(REPO_PATH)/vendor/github.com/golang/protobuf/protoc-gen-go
 
 bin/golint:
-	@go install -v $(REPO_PATH)/vendor/golang.org/x/lint/golint
+	@go install -v $(THIS_DIRECTORY)/vendor/golang.org/x/lint/golint
 
 clean:
 	@rm -rf bin/

